src/regionprops.jl

```diff
@@ -202,12 +246,67 @@ function cropfloe(floesimg::BitMatrix, props::DataFrame, i::Int64)
     return prefloe
 end
 
+"""
+    cropfloe(floesimg, min_row, min_col, max_row, max_col, label)
+
+Crops the floe from `floesimg` with the label `label`, returning the region bounded by `min_row`, `min_col`, `max_row`, `max_col`, and converting to a BitMatrix.
+"""
+function cropfloe(floesimg::Matrix{I}, min_row::J, min_col::J, max_row::J, max_col::J, label::I)  where {I<:Integer, J<:Integer}
+    #= 
+    Crop the floe using bounding box data in props.
+    Note: Using a view of the cropped floe was considered but if there were multiple components in the cropped floe, the source array with the floes would be modified. =#
+    prefloe = floesimg[min_row:max_row, min_col:max_col]
+    @debug "prefloe: $prefloe"
+
+    #= Remove any pixels not corresponding to that numbered floe 
+    (each segment has a different integer) =#
+    floe_area = prefloe .== label
+    @debug "mask: $floe_area"
+
+    return floe_area
+end
+
+"""
+    cropfloe(floesimg, label)
+
+Crops the floe from `floesimg` with the label `label`, adding a one pixel border of zeros and converting to a BitMatrix.
+"""
+function cropfloe(floesimg::Matrix{I}, label::I) where {I<:Integer}
+    #= Remove any pixels not corresponding to that numbered floe 
+    (each segment has a different integer) =#
+    floe_area = floesimg .== label
+    @debug "mask: $floe_area"
+
+    # Crop the floe to the size of the floe.
+    nonzero = x -> x > 0
+    rows = 2
+    cols = 1
+    row_sums = count(floe_area, dims=rows)
+    col_sums = count(floe_area, dims=cols)
+    @debug "row_sums: $row_sums, col_sums: $col_sums"
+
+    min_row = findfirst(nonzero, row_sums)[cols]
+    min_col = findfirst(nonzero, col_sums)[rows]
+    max_row = findlast(nonzero, row_sums)[cols]
+    max_col = findlast(nonzero, col_sums)[rows]
+    @debug "($min_row, $min_col), ($max_row, $max_col)"
```

Comment:
This is interesting! Consider moving it to a function of its own (`get_bbox`?).

Reply:
It would pretty much be the whole function, from line 286 through 297, and I'm not sure we use it anywhere else, so I don't know that abstracting it would make the code more readable. Is there somewhere else we might call it? The old way of cropping the floes (with `bwareamaxfilt`) is subtly different, so we can't reuse it there.

Reply:
This has been on my mind for the last couple of days. I recalled that individual floe masks are needed for the tracker. One is able to get all of them in one go with the [`bbox` prop from `regionprops`](https://scikit-image.org/docs/stable/api/skimage.measure.html#skimage.measure.regionprops:~:text=as%20the%20region.-,bboxtuple,-Bounding%20box%20(min_row ) as I did [here](https://github.com/WilhelmusLab/IceFloeTracker.jl/blob/939866c522f0e27c86185a11a2017634b5e6ac75/src/regionprops.jl#L87). Did you consider that?

Reply:
I think if we do that, we should avoid adding a new python dependency, and instead use something like https://github.com/zygmuntszpak/ImageComponentAnalysis.jl/blob/0d4503f5a4b2bed647f99a40adc485d5413c8afb/src/algorithms/bounding_box.jl#L1-L26

But do we want to do that now? It looks to be doing the same thing, but it's assuming it's going to loop over a bunch of labels, rather than providing us the ability to do just one. Could we put this in as an issue and come back to it later?

Reply:
The way that I see this is that one would be processing one now but later one would like to aggregate them all for the tracker step. Getting the bbox from regionprops avoids that extra overhead entirely.

Reply:
@mirestrepo It's from an existing dependency (`skimage`). We've had conversations around how to eliminate it but it's not easy as there are props we need for the tracker that we can't get from the Julia ecosystem without extra work.

Reply:
After some checks, I've removed this bit of code, as we won't need it.